src/views/Nft/market/Collection/Items/CollectionNfts.tsx

```diff
@@ -17,33 +22,65 @@ interface CollectionNftsProps {
 
 const CollectionNfts: React.FC<CollectionNftsProps> = ({ collection, sortBy = 'updatedAt' }) => {
   const { address } = collection
+  const [allPancakeBunnyNfts, setAllPancakeBunnyNfts] = useState<NftToken[]>(null)
   const checksummedAddress = getAddress(address)
   const nfts = useNftsFromCollection(checksummedAddress)
   const dispatch = useAppDispatch()
 
+  const isPBCollection = address === pancakeBunniesAddress
+
   useEffect(() => {
     dispatch(fetchNftsFromCollections(checksummedAddress))
   }, [checksummedAddress, dispatch])
 
-  if (!nfts) {
+  useEffect(() => {
+    const fetchPancakeBunnies = async () => {
+      // In order to not define special TS type just for PancakeBunnies display we're hacking a little bit into NftToken type.
+      // On this page we just want to display all bunnies with thier lowest prices and updates on the market
+      // Since some bunnies might not be on the market at all, we don't refer to the redux nfts state (which stores NftToken with actual token ids)
+      // We merely request from API all available bunny ids with their metadata and query subgraph for lowest price and latest updates.
+      const { data } = await getNftsFromCollectionApi(pancakeBunniesAddress)
+      const bunnyIds = Object.keys(data)
+      const lowestPrices = await getAllPancakeBunniesLowestPrice(bunnyIds)
+      const latestUpdates = await getAllPancakeBunniesRecentUpdatedAt(bunnyIds)
+      const allBunnies: NftToken[] = bunnyIds.map((bunnyId) => {
+        return {
+          // tokenId here is just a dummy one to satisfy TS. TokenID does not play any role in gird display below
+          tokenId: data[bunnyId].name,
```

Comment:
You can just type `allBunnies`
```
Omit<NftToken, 'tokenId'>[]
```

Reply:
@hachiojidev I don't think I can, its passed to components as props and they expect `NftToken`.
I can modify prop type to `NftToken | Omit<NftToken, 'tokenId'>` though 🤔

Reply:
Yeah good point, didn't consider that. Not ideal but best to fake the token id. Or...making the `tokenId` optional.